app/src/main/java/com/glia/exampleapp/Application.java

```diff
@@ -8,6 +15,38 @@ public class Application extends android.app.Application {
     public void onCreate() {
         super.onCreate();
         GliaWidgets.onAppCreate(this);
+
+        initGliaWidgets();
     }
 
+    private void initGliaWidgets() {
+        registerActivityLifecycleCallbacks(new ActivityLifecycleCallbacks() {
+            @Override
+            public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState) {
```

Comment:
I would add some comment to eplain why it is done like that. Otherwise it might be very confusing if you don't know about the deep link logic inside the main activity